### PR TITLE
Add remote enclave examples

### PR DESCRIPTION
**Pull request by prd-fox**
This updates the `Vagrantfile` and `docker-compose` file to use Tessera v0.9.

It also contains 7node examples of using Tessera with a remote enclave - one with all 7 nodes using an enclave and another which defaults to 4, but can be changed for any number from 1 to 7.

### CODE REVIEW

**Comment by prd-fox** [comment]
@fixanoid I merely copied the 7nodes and edited the scripts to start Tessera with a remote enclave. Do you reckon it should be changed to have some sort of structure like:

-> 7nodes
----> local-enclave
----> remote-enclave
----> mixed-enclave

so that all the duplication of key files etc can be removed, or keep it as is as distinct examples?

**Comment by fixanoid** [comment]
@prd-fox I think having distinct examples where more than 90% of contents is the same is an overkill. Lets keep the single 7nodes example we always had, but add new option / shell starters for there distinct examples.

**Comment by prd-fox** [comment]
@fixanoid I've updated it to be a lot cleaner now. PTAL.

**Comment by prd-fox** [comment]
@SatpalSandhu61 gave some offline comments which I will address soon

File: examples/7nodes/README.md [hunk 1]
```diff
@@ -216,3 +216,12 @@ After making these changes, the `raft-init.sh` and `raft-start.sh` scripts can b
     </pre>
 
 After saving this change, the `./runscript.sh private-contract.js` command can be run as usual to submit the private contract.  You can then follow steps described above to verify that node 5 can see the transaction payload and that nodes 2-4 are unable to see the payload.
+
+## Using a Tessera remote enclave
+Tessera v0.9 brought with it the option to have an enclave as a separate process from the Transaction
+Manager. This is a more secure way of being able to manage and interact with your keys.
+To use the remote enclave, call ypur desired start with using `tessera-remote` as the first
```
Comment (fixanoid) [inline examples/7nodes/README.md]: "your"

**Review by fixanoid — APPROVED** [review]
❤️

**Pull request merged**